docs/docs/services/qix-engine/access-control.md

```diff
@@ -1,76 +1,135 @@
+makes it possible to control the document resources based on users' attributes rather than using roles or other more
+static means.
 
-## Configuration
+## Rules
 
-Rules are defined, one on each row, in text files. Qlik Associative Engine will
-apply rules in order (except for deny rules which are always applied first,
-regardless of where they are defined).
+Each rule is a conditional expression that evaluates to either `true` or `false`. In addition to this, each rule
+contains a mandatory part that determines which actions the rule grants, or denies.
 
-If the ABAC feature is enabled, all users will by default have no
-access, so it is important that you also include rules when enabling it
-or you will cause access problems for your users.
+!!! Note
+    If the ABAC feature is enabled, all users will by default have no access, so it is important to also include
+    rules when enabling it. Otherwise, access problems for users can easily be caused.
+
+### Rules Overview
+
+Upcoming sections more formally describe how rules are formed, but to get an initial understanding, consider the
+following rule:
+
+```c
+user.id = "ada-lovelace" and resource._resourcetype = "App" and resource._actions = {"create", "update", "read"}
+```
+
+This rule states that if the user is `ada-lovelace` and the accessed resource type is `"App"`, grant the actions
+`create`, `update`, and `read`. The `user.id` and `resource._resourcetype` expressions could be omitted and the rule
+would still be valid.
+
+The expression involving `_actions` must be present in order to determine which actions that are granted access. See
+section [Actions](#actions) for more details.
+
+### Rule Files
+
+Rules are defined in text files with one rule on each row. Qlik Associative Engine applies rules in the order they
+appear in the file. Two types of rule files are supported. The _Deny_ rule file and the _Allow_ rule file.
+
+The Deny rule file denies access and is always evaluated first. If any rule in the Deny file evaluates to `true`,
+access is immediately denied for the actions provided and no further evaluation of rules take place.
+
+The Allow rule file allows access. If a rule in the Allow file evaluates to `true`, access is granted for the
+actions specified and rule evaluation continues, possibly accumulating access to more actions based on succeeding
+rules.
+
+!!! Note
+    The rule files are read by Qlik Associative Engine at start. Modifying the rule files requires a restart in order
+    for new rules to take effect.
+
+## Engine Configuration for ABAC
+
+### ABAC-related Command Line Switches
+
+| Switch | Values | Default | Description |
+| ------ | ------ | ------- | ----------- |
+| `EnableABAC` | `0` or `1` | `0` | Enabling/disabling of ABAC rule evalutation. |
+| `Gen3` | `0` or `1` | `0` | Enablig/disabling of server mode. Must be set to `1` if enabling ABAC. |
```

Comment:
Not sure we should expose the Gen3 flag. I think we should enabled it internally and implicitly through the EnableABAC flag.